index.js

```diff
@@ -1,31 +1,32 @@
 'use strict';
-var os = require('os');
-var macosRelease = require('macos-release');
-var winRelease = require('win-release');
+const os = require('os');
+const macosRelease = require('macos-release');
+const winRelease = require('windows-release');
 
 module.exports = function (platform, release) {
 	if (!platform && release) {
 		throw new Error('You can\'t specify a `release` without specifying `platform`');
 	}
 
 	platform = platform || os.platform();
-	release = release || os.release();
 
-	var id;
+	let id;
 
 	if (platform === 'darwin') {
-		var prefix = Number(release.split('.')[0]) > 15 ? 'macOS' : 'OS X';
+		release = release || os.release();
+		const prefix = Number(release.split('.')[0]) > 15 ? 'macOS' : 'OS X';
 		id = macosRelease(release).name;
 		return prefix + (id ? ' ' + id : '');
 	}
 
 	if (platform === 'linux') {
+		release = release || os.release();
 		id = release.replace(/^(\d+\.\d+).*/, '$1');
 		return 'Linux' + (id ? ' ' + id : '');
 	}
 
 	if (platform === 'win32') {
-		id = winRelease(release);
+		id = release ? winRelease(release) : '';
```

Comment:
It should still return `Windows 10` if calling `osName()` or `osName('win32')` (notice no arguments) on Windows 10.

Reply:
So fuzzy... So you want:

When running in the same os than the queried, return full os + release name

Otherwise, return the is only

Right?